src/openvic-simulation/economy/FactoryProducer.cpp

```diff
@@ -0,0 +1,29 @@
+#include "FactoryProducer.hpp"
+
+using namespace OpenVic;
+
+CTORCPP(
+	FactoryProducer, ProductionType const&, production_type, const fixed_point_t, size_multiplier, const fixed_point_t,
+	revenue_yesterday, const fixed_point_t, output_quantity_yesterday, const fixed_point_t, unsold_quantity_yesterday,
+	FactoryProducer::employees_map_t&&, employees, Good::good_map_t&&, stockpile, fixed_point_t, budget, const fixed_point_t,
+	balance_yesterday, const fixed_point_t, received_investments_yesterday, const fixed_point_t, market_spendings_yesterday,
+	const fixed_point_t, paychecks_yesterday, const uint32_t, unprofitable_days, const uint32_t, injected_days, const uint32_t,
+	days_without_input, const uint8_t, hiring_priority, const uint8_t, profit_history_current,
+	FactoryProducer::daily_profit_history_t&&, daily_profit_history
+);
+FactoryProducer::FactoryProducer(ProductionType const& new_production_type, const fixed_point_t new_size_multiplier)
+	: FactoryProducer(new_production_type, new_size_multiplier, 0, 0, 0, {}, {}, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, {}) {}
```

Comment:
```suggestion
	: FactoryProducer { new_production_type, new_size_multiplier, 0, 0, 0, {}, {}, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, {} } {}
```
In general, [curly brackets should be preferred to round brackets](https://stackoverflow.com/a/18222927) for initialising things, round brackets allow certain implicit type conversions that can hide errors.
`ResourceGatheringOpertion`'s second constructor should also use them.